org.jacoco.core.test/src/org/jacoco/core/test/validation/StatementParser.java

```diff
@@ -0,0 +1,133 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Marc R. Hoffmann - initial API and implementation
+ *    
+ *******************************************************************************/
+package org.jacoco.core.test.validation;
+
+import java.io.IOException;
+import java.io.StreamTokenizer;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Simple parser for Java like method invocation statements. For example:
+ * 
+ * <pre>
+ * foo("BAR", 42);
+ * </pre>
+ * 
+ * Method parameters can only be String or int literals.
+ */
+public class StatementParser {
+
+	/**
+	 * Call-back interface for parsed statements.
+	 */
+	public interface IStatementVisitor {
+
+		/**
+		 * Called when a method invocation has been paresed.
+		 * 
+		 * @param ctx
+		 *            context information for error messages
+		 * @param name
+		 *            local method name
+		 * @param args
+		 *            argument values
+		 */
+		void visitInvocation(String ctx, String name, Object... args);
+
+	}
+
+	/**
+	 * Parses the given source.
+	 * 
+	 * @param source
+	 *            source string to parse
+	 * @param visitor
+	 *            visitor to emit parsed statements
+	 * @param ctx
+	 *            context information to include in error messages
+	 */
+	public static void parse(String source, IStatementVisitor visitor,
+			String ctx) throws IOException {
+		new StatementParser(source, visitor, ctx).parse();
+	}
+
+	private final IStatementVisitor visitor;
+	private final StreamTokenizer tokenizer;
+	private final String ctx;
+
+	private StatementParser(String source, IStatementVisitor visitor,
+			String ctx) {
+		this.visitor = visitor;
+		this.ctx = ctx;
+		tokenizer = new StreamTokenizer(new StringReader(source));
+		tokenizer.resetSyntax();
+		tokenizer.whitespaceChars(' ', ' ');
+		tokenizer.whitespaceChars('\t', '\t');
+		tokenizer.wordChars('a', 'z');
+		tokenizer.wordChars('A', 'Z');
+		tokenizer.quoteChar('"');
+		tokenizer.parseNumbers();
+	}
+
+	private void parse() throws IOException {
+		while (!accept(StreamTokenizer.TT_EOF)) {
+			invocation();
+		}
+	}
+
+	private void invocation() throws IOException {
+		final String name = expect(StreamTokenizer.TT_WORD).sval;
+		final List<Object> args = new ArrayList<Object>();
+		expect('(');
+		if (!accept(')')) {
+			args.add(argument());
+			while (!accept(')')) {
+				expect(',');
+				args.add(argument());
+			}
+		}
+		expect(';');

```

Comment:
To me seems that removal of `;` is safe. This will make tests cleaner. After replacement on
```
		if (tokenizer.nextToken() != ';') {
			tokenizer.pushBack();
		}
```
all tests except `StatementParserTest.should_fail_when_semicolon_is_missing` pass.

Reply:
My intention was to use Java syntax. What about multiple invocations in a single comment? Should we require a colon here as a separator?

Reply:
Colon (`:`) sounds/looks even more weird than current presence of semicolon.

Semicolon IMO really redundant in case of single invocation. And why not omit separator in case of multiple invocations? Still will be Java-like syntax

```diff
--- org.jacoco.core.test.validation.java5/src/org/jacoco/core/test/validation/java5/targets/FinallyTarget.java
+++ org.jacoco.core.test.validation.java5/src/org/jacoco/core/test/validation/java5/targets/FinallyTarget.java
@@ -75,11 +75,11 @@
 	private static void catchNotExecuted() {
 		try {
 			nop("try");
-		} catch (Exception e) { // > tag("catchNotExecuted");
-			nop("catch"); // > assertNotCovered();
-		} finally { // > assertEmpty();
-			nop("finally"); // >assertFullyCovered();tag("catchNotExecuted.1");
-		} // > assertEmpty(); tag("catchNotExecuted.2");
+		} catch (Exception e) { // tag("catchNotExecuted")
+			nop("catch"); // assertNotCovered()
+		} finally { // assertEmpty()
+			nop("finally"); // assertFullyCovered() tag("catchNotExecuted.1")
+		} // assertEmpty() tag("catchNotExecuted.2")
 	}
 
 	private static void emptyCatch() {
```

Reply:
@Godin Sorry, I wanted to say *semicolon*.

Here is another catch: For long lines we must avoid whitespaces, otherwise the auto-formatter wraps the commend into multiple lines. In this case the comment would read:

    nop("finally"); // assertFullyCovered()tag("catchNotExecuted.1")

If this is ok for you I would remove semicolon from the syntax at all.

Reply:
✔️ Semicolons are gone.

Reply:
BTW, I found two instances where I already forgot to include `>` 😉

Reply:
I would not say that semicolon of a much help for formatting, so 👍 about removal.

About forgotten `>` see https://github.com/jacoco/jacoco/pull/718#discussion_r209002552 - I still don't know how you found second one 😆